lib/rules/no-class.js

```diff
@@ -0,0 +1,27 @@
+/**
+ * @fileoverview Avoid using class inheritance, instead favor prototypes or object composition.
+ * @author Nicola Molinari
+ * @copyright 2015 Nicola Molinari. All rights reserved.
+ */
+"use strict";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = function(context) {
+
+    //--------------------------------------------------------------------------
+    // Public
+    //--------------------------------------------------------------------------
+
+    return {
+
+        "ClassDeclaration": function(node) {
+            context.report(node, "Unexpected class declaration (with ecmaFeatures.classes enabled)" +
+              ", you should use other methods like prototypal inheritance or object composition.");
```

Comment:
I think just `"Unexpected class declaration."` would be sufficient. The exact reasoning, why it is considered a problem should be explained in the rule documentation.